File: apps/cron/src/jobs/invitations.ts
import type { Database } from "@raven/db";
import { invitations } from "@raven/db";
import { and, isNull, lt } from "drizzle-orm";

export const cleanupExpiredInvitations = async (db: Database) => {
  const now = new Date();

  const deleted = await db
    .delete(invitations)
    .where(and(lt(invitations.expiresAt, now), isNull(invitations.acceptedAt)))
    .returning({ id: invitations.id });

  console.log(
    `Invitation cleanup: deleted ${deleted.length} expired invitations`
  );
};

File: apps/cron-py/src/jobs/invitations.py
def cleanup_expired_invitations(conn) -> None:
    with conn.cursor() as cur:
        cur.execute("""
            DELETE FROM invitations
            WHERE "expiresAt" < NOW()
              AND "acceptedAt" IS NULL
        """)
        count = cur.rowcount
        if count > 0:
            print(f"  Deleted {count} expired invitation(s)")
